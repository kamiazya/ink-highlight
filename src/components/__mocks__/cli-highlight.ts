import { HighlightOptions } from 'cli-highlight';

const j = JSON.stringify;

export const highlight = jest.fn((code: string, options?: HighlightOptions): string => {
  if (options) {
    return `highlight(${j(code)}, ${j(options)})`;
  }
  return `highlight(${j(code)})`;
});
