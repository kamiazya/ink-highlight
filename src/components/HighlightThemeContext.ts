import { Theme as ITheme, DEFAULT_THEME } from 'cli-highlight';
import { createContext } from 'react';

export const HighlightThemeContext = createContext<ITheme>(DEFAULT_THEME);
