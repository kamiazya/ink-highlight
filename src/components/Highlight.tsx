import React, { FC, useContext } from 'react';
import PropTypes from 'prop-types';
import { highlight } from 'cli-highlight';
import { HighlightOptions, HighlightOptionsContext } from './HighlightOptionsContext';
import { HighlightThemeContext } from './HighlightThemeContext';

export type Props = {
  code: string;
} & HighlightOptions;

export const Highlight: FC<Props> = ({ code, ...option }) => {
  const theme = useContext(HighlightThemeContext);
  const contextOption = useContext(HighlightOptionsContext);
  return (
    <>
      {highlight(code, {
        ...contextOption,
        ...option,
        theme,
      })}
    </>
  );
};

Highlight.propTypes = {
  code: PropTypes.string.isRequired,
  language: PropTypes.string,
  ignoreIllegals: PropTypes.bool,
  continuation: PropTypes.any,
  languageSubset: PropTypes.arrayOf(PropTypes.string.isRequired),
};

Highlight.defaultProps = {
  language: undefined,
  ignoreIllegals: undefined,
  continuation: undefined,
  languageSubset: undefined,
};
