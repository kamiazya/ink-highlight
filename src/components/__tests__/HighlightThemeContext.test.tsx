import React from 'react';
import { wrap } from 'jest-snapshot-serializer-raw';
import { render } from 'ink-testing-library';
import { Theme } from 'cli-highlight';
import { Highlight } from '../Highlight';
import { HighlightThemeContext } from '../HighlightThemeContext';

describe('HighlightThemeContext', () => {
  test('Theme applied', () => {
    const theme = Object.freeze({ test: true }) as Theme;
    const { lastFrame } = render(
      <HighlightThemeContext.Provider value={theme}>
        <Highlight code="console.log('Hello world!');" />
      </HighlightThemeContext.Provider>,
    );
    expect(wrap(lastFrame() as string)).toMatchSnapshot();
  });
});
