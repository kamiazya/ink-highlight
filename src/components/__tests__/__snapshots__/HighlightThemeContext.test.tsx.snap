// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HighlightThemeContext Theme applied 1`] = `highlight("console.log('Hello world!');", {"theme":{"test":true}})`;
